tests/unit/payment-methods/test-class-digital-wallets-payment-gateway.php

```diff
@@ -1,158 +0,0 @@
-	 */
-	private $mock_token_service;
-
-	/**
-	 * Mock WC_Payments_API_Client.
-	 *
-	 * @var WC_Payments_API_Client|PHPUnit_Framework_MockObject_MockObject
-	 */
-	private $mock_api_client;
-
-	/**
-	 * Mock WC_Payments_Action_Scheduler_Service.
-	 *
-	 * @var WC_Payments_Action_Scheduler_Service|PHPUnit_Framework_MockObject_MockObject
-	 */
-	private $mock_action_scheduler_service;
-
-	/**
-	 * WC_Payments_Account instance.
-	 *
-	 * @var WC_Payments_Account
-	 */
-	private $wcpay_account;
-
-	/**
-	 * Mocked value of return_url.
-	 * The value is used in the set up and tests, so it's set as a private
-	 * variable for easy reference.
-	 *
-	 * @var string
-	 */
-	private $return_url = 'test_url';
-
-	public function setUp() {
-		parent::setUp();
-
-		// Arrange: Mock WC_Payments_API_Client so we can configure the
-		// return value of create_and_confirm_intention().
-		// Note that we cannot use createStub here since it's not defined in PHPUnit 6.5.
-		$this->mock_api_client = $this->getMockBuilder( 'WC_Payments_API_Client' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'create_and_confirm_intention', 'get_payment_method', 'is_server_connected' ] )
-			->getMock();
-
-		// Arrange: Create new WC_Payments_Account instance to use later.
-		$this->wcpay_account = new WC_Payments_Account( $this->mock_api_client );
-
-		// Arrange: Mock WC_Payments_Customer_Service so its methods aren't called directly.
-		$this->mock_customer_service = $this->getMockBuilder( 'WC_Payments_Customer_Service' )
-			->disableOriginalConstructor()
-			->getMock();
-
-		// Arrange: Mock WC_Payments_Customer_Service so its methods aren't called directly.
-		$this->mock_token_service = $this->getMockBuilder( 'WC_Payments_Token_Service' )
-			->disableOriginalConstructor()
-			->getMock();
-
-		// Arrange: Mock WC_Payments_Action_Scheduler_Service so its methods aren't called directly.
-		$this->mock_action_scheduler_service = $this->getMockBuilder( 'WC_Payments_Action_Scheduler_Service' )
-			->disableOriginalConstructor()
-			->getMock();
-
-		$this->mock_wcpay_gateway = $this->getMockBuilder( Digital_Wallets_Payment_Gateway::class )
-			->setConstructorArgs(
-				[
-					$this->mock_api_client,
-					$this->wcpay_account,
-					$this->mock_customer_service,
-					$this->mock_token_service,
-					$this->mock_action_scheduler_service,
-				]
-			)
-			->setMethods( [ 'get_option' ] )
-			->getMock();
-	}
-
-	/**
-	 * Test first time setting up such that digital_wallets_enabled_locations is not set. It should return defaults.
-	 */
-	public function test_get_digital_wallets_enabled_locations_defaults() {
-		$this->mock_wcpay_gateway->expects( $this->once() )
-			->method( 'get_option' )
-			->with( 'digital_wallets_enabled_locations' )
-			->willReturn( '' );
-
-		$actual   = $this->mock_wcpay_gateway->get_digital_wallets_enabled_locations();
-		$expected = [
-			'cart'         => true,
-			'checkout'     => true,
-			'product_page' => true,
-		];
```

Comment:
This test is testing the regression mentioned above. Maybe we can backport this test to use the `payment_request_button_locations` option from WCPay_Gateway to prevent such situations?

Reply:
@frosso What do you think about this test? Should we backport it?

Reply:
Good idea - I did it in 4e5644a7